plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergDynamicPartitionPruningTest.java

```diff
@@ -66,9 +60,27 @@ protected void createPartitionedTable(String tableName, List<String> columns, Li
         getQueryRunner().execute(sql);
     }
 
+    @Override
+    protected void createPartitionedAndBucketedTable()
+    {
+        // Use decimal type because Iceberg doesn't allow float type on bucket columns
+        createPartitionedAndBucketedTable("t0_bucketed", ImmutableList.of("v0 decimal(2, 1)", "k0 integer"), ImmutableList.of("k0"), ImmutableList.of("v0"));
+        createPartitionedAndBucketedTable("t1_bucketed", ImmutableList.of("v1 decimal(2, 1)", "i1 integer"), ImmutableList.of(), ImmutableList.of("v1"));
+        createPartitionedAndBucketedTable("t2_bucketed", ImmutableList.of("v2 decimal(2, 1)", "i2 integer", "k2 integer"), ImmutableList.of("k2"), ImmutableList.of("v2"));
+    }
+
     @Override
     protected void createPartitionedAndBucketedTable(String tableName, List<String> columns, List<String> partitionColumns, List<String> bucketColumns)
     {
-        throw new UnsupportedOperationException();
+        ImmutableList.Builder<String> partitioning = ImmutableList.builder();
+        partitionColumns.forEach(partitioning::add);
+        bucketColumns.forEach(column -> partitioning.add("bucket(%s,10)".formatted(column)));
+
+        @Language("SQL") String sql = format(
```

Comment:
`@Language("SQL")` is redundant (remove), since IntelliJ can infer based on usage
and it's also useless, since `format` prevents IntelliJ from syntax highlighting